### PR TITLE
Update bundle to 2.12.0

### PR DESCRIPTION
**Pull request by dbartol**
### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Comment by angelapwen** [comment]
Hi — I've just merged https://github.com/github/codeql-action/pull/1463 which creates some merge conflicts here. We'll be updating the release documentation shortly to include this, but wanted to write an FYI that we've added a few fields to the file that include the actual version number of the current release, as well as the bundle name + version number of the previous release 😄 let me know if any questions!

**Comment by dbartol** [comment]
@angelapwen Thanks for the heads up!

**Review by henrymercer — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## [UNRELEASED]
 
-No user facing changes.
+- Update default CodeQL bundle version to 2.12.0. [#1466](https://github.com/github/codeql-action/pull/1466)
```
Comment (aeisenberg) [inline CHANGELOG.md:5]: Something minor and doesn't need to happen for this release, but perhaps we should start linking to the release notes for the CLI version we are upgrading to.

**Review by aeisenberg — APPROVED** [review]
To be clear...this integration test will continue to fail until it starts using a version of the CLI that supports kotlin 1.8.0?

I think that is fine, but if this gets annoying, we should disable this test in main and create an issue to remind ourselves to re-enable it.

**Review by adityasharad — APPROVED** [review]

**Pull request merged**